android/src/main/java/io/radar/capacitor/RadarPlugin.java

```diff
@@ -637,30 +751,29 @@ public void onComplete(@NotNull Radar.RadarStatus status, @Nullable RadarRoutes
         destination.setLongitude(destinationLongitude);
         destination.setAccuracy(5);
 
-        if (!call.hasOption("modes")) {
-            call.reject("modes is required");
-
-            return;
-        }
         EnumSet<Radar.RadarRouteMode> modes = EnumSet.noneOf(Radar.RadarRouteMode.class);
-        List<String> modesList = call.getArray("modes").toList();
-        if (modesList.contains("FOOT") || modesList.contains("foot")) {
-            modes.add(Radar.RadarRouteMode.FOOT);
-        }
-        if (modesList.contains("BIKE") || modesList.contains("bike")) {
-            modes.add(Radar.RadarRouteMode.BIKE);
-        }
-        if (modesList.contains("CAR") || modesList.contains("car")) {
-            modes.add(Radar.RadarRouteMode.CAR);
+        List<String> modesList = new ArrayList<>(call.getArray("modes").toList());
+        for (Radar.RadarRouteMode mode : Radar.RadarRouteMode.values()) {
+            int index = modesList.indexOf(mode.name());
+            if (index > 0) {
+                modes.add(mode);
+                modesList.remove(index);
```

Comment:
It's unlikely, but vaguely wondering if this would be a `ConcurrentModificationException`